oak_runtime/src/node/http/client.rs

```diff
@@ -0,0 +1,291 @@
+            debug!("Waiting for HTTP invocation");
+            // Read an HTTP invocation from the [`Receiver`].
+            let invocation = receiver.receive(&runtime).map_err(|error| {
+                match error {
+                    OakError::OakStatus(OakStatus::ErrTerminated) => {
+                        debug!("HTTP client node is terminating.")
+                    }
+                    OakError::OakStatus(OakStatus::ErrChannelClosed) => {
+                        info!("HTTP invocation channel closed")
+                    }
+                    _ => error!("Couldn't receive the invocation: {:?}", error),
+                }
+                error
+            })?;
+
+            let result = self.process_invocation(&runtime, &invocation).await;
+            info!("Invocation processing finished: {:?}", result);
+            if result.is_err() {
+                warn!("Error encountered: {:?}", result);
+            }
+            invocation.close(&runtime);
+        }
+    }
+
+    /// Process an HTTP invocation for an external HTTP service.
+    async fn process_invocation(
+        &mut self,
+        runtime: &RuntimeProxy,
+        invocation: &Invocation,
+    ) -> Result<(), OakError> {
+        // TODO(#1717): Update HTTP client metrics
+
+        // Define a closure to ease logging the errors
+        let send_error = |code, msg| {
+            invocation.send_error(code, msg, runtime);
+        };
+
+        // Receive a request from the invocation channel.
+        let request = invocation.receive_request(&runtime).map_err(|error| {
+            send_error(
+                http::StatusCode::INTERNAL_SERVER_ERROR,
+                "Failed to read request",
+            );
+            error!(
+                "Couldn't read HTTP request from the invocation: {:?}",
+                error
+            );
+            error
+        })?;
+        debug!("Incoming HTTP request: {:?}", request);
+
+        let mut hyper_request_builder = hyper::Request::builder()
+            .method(&request.method[..])
+            .uri(request.uri);
+
+        if let Some(headers) = request.headers {
+            for (header_name, header_value) in headers.into_iter() {
+                hyper_request_builder = hyper_request_builder.header(header_name, header_value);
+            }
+        }
+        let hyper_request = hyper_request_builder
+            .body(hyper::Body::from(request.body))
+            .map_err(|err| {
+                warn!("Error when processing the request: {:?}", err);
+                OakError::OakStatus(OakStatus::ErrInternal)
+            })?;
+
+        // Forward the request to the external HTTP service and wait for the response.
+        let resp = self
+            .http_client
+            .request(hyper_request)
+            .await
+            .map_err(|error| {
+                error!("Error while awaiting response: {}", error);
+                // TO-DO: Retry instead?
+                send_error(
```

Comment:
nit: this seems quite inconsistent. If the body parsing fails above, it won't send an error back, and only logs a warning. Here it logs an error and sends an error response back (which logs another error). At the point where the response is handled, another warning is logged. What is the criteria for logging errors vs warning and sending an error response back vs not?